src/main/plugin/iso19139.ca.HNAP/layout/tpl-csv.xsl

```diff
@@ -0,0 +1,199 @@
+          <xsl:otherwise>dataset</xsl:otherwise>
+        </xsl:choose>
+      </category>
+      <metadatacreationdate>
+        <xsl:value-of select="gmd:dateStamp/*"/>
+      </metadatacreationdate>
+      <referencesystem>
+        <xsl:value-of select="gmd:referenceSystemInfo/gmd:MD_ReferenceSystem/gmd:referenceSystemIdentifier/gmd:RS_Identifier/gmd:code/gco:CharacterString"/>
+      </referencesystem>
+
+      <!-- For each date grouped by the date type mapped to a readable value -->
+      <xsl:for-each select="gmd:identificationInfo/*/gmd:citation/*/gmd:date">
+        <xsl:variable name="dateTypeCode" select="*/gmd:dateType/*/@codeListValue"/>
+        <xsl:variable name="dateTypeCodeReadable" select="tokenize($codelists/codelist[@name = 'gmd:CI_DateTypeCode']/entry[code/text() = $dateTypeCode]/value/text(), ';')[1]"/>
+
+        <xsl:element name="date-{$dateTypeCodeReadable}">
+          <xsl:value-of select="*/gmd:date/*/text()"/>
+        </xsl:element>
+      </xsl:for-each>
+
+      <xsl:for-each select="gmd:identificationInfo/*/gmd:graphicOverview/*/gmd:fileName">
+        <image>
+          <xsl:value-of select="*/text()"/>
+        </image>
+      </xsl:for-each>
+
+      <!-- All keywords not having thesaurus reference or an empty thesaurusName -->
+      <xsl:for-each select="gmd:identificationInfo/*/gmd:descriptiveKeywords/gmd:MD_Keywords[not(gmd:thesaurusName) or gmd:thesaurusName/*/gmd:identifier/*/gmd:code/*/text() = '']">
+        <xsl:variable name="keywordTypeCode" select="gmd:type/*/@codeListValue"/>
+        <xsl:variable name="keywordTypeCodeReadable" select="tokenize($codelists/codelist[@name = 'gmd:MD_KeywordTypeCode']/entry[code/text() = $keywordTypeCode]/value/text(), ';')[1]"/>
+
+        <xsl:for-each select="gmd:keyword[not(@gco:nilReason) or */text() != '']">
+          <xsl:choose>
+            <!-- All keywords with a valid keywordTypeCode -->
+            <xsl:when test="$keywordTypeCodeReadable != ''">
+              <xsl:element name="keyword-{$keywordTypeCodeReadable}">
+                <xsl:apply-templates mode="localised" select=".">
+                  <xsl:with-param name="langId" select="$langId"/>
+                </xsl:apply-templates>
+              </xsl:element>
+            </xsl:when>
+
+            <!-- All keywords without a valid keywordTypeCode -->
+            <xsl:otherwise>
+              <keyword-other>
+                <xsl:apply-templates mode="localised" select=".">
+                  <xsl:with-param name="langId" select="$langId"/>
+                </xsl:apply-templates>
+              </keyword-other>
+            </xsl:otherwise>
+          </xsl:choose>
+        </xsl:for-each>
+      </xsl:for-each>
+
+      <!-- All keywords with a valid thesaurus name -->
+      <xsl:for-each select="gmd:identificationInfo/*/gmd:descriptiveKeywords/*[gmd:thesaurusName/*/gmd:identifier/*/gmd:code/*/text() != '']">
```

Comment:
It is a little odd to do the loop based on `gmd:thesaurusName/*/gmd:identifier/*/gmd:code/*/text() != '`' for one loop and `gmd:thesaurusName/*/gmd:identifier/*/gmd:code/*/text() = ''` for the other loop. When it has the following

```
<xsl:variable name="thesaurusId" select="normalize-space(gmd:thesaurusName/*/gmd:identifier/*/gmd:code/*/text())"/>
<xsl:variable name="thesaurusKey" select="replace($thesaurusId, '[^a-zA-Z0-9]', '')"/>
```

i.e. lets assume the `gmd:code/*/text() = "@"`
And for another thesauruse the `gmd:code/*/text() = "$"`

In this case the thesaurusId will be '' for both and cause unique conflict errors.

I woule expect 
`[replace(gmd:thesaurusName/*/gmd:identifier/*/gmd:code/*/text(), '[^a-zA-Z0-9]') = '']`
and 
`[replace(gmd:thesaurusName/*/gmd:identifier/*/gmd:code/*/text(), '[^a-zA-Z0-9]') != '']`

It is not critical as I can't see where someone would only use none alpha chars in the value.